hand-eye-calibration/full_hand_eye_sample.py

```diff
@@ -0,0 +1,510 @@
+"""
+Script to generate a dataset and perform hand-eye calibration using a Universal Robot UR5e robot.
+The script communicates with the robot through TCP/IP.
+
+The entire sample consist of two additional files:
+    - universal_robots_hand_eye_script.urp: Robot program script that moves between different poses.
+    - robot_communication_file.xml: communication set-up file.
+
+Running the sample requires that you have universal_robots_hand_eye_script.urp on your UR5e robot,
+and robot_communication_file.xml in the same repo as this sample. Each robot pose
+must be modified to your scene. This is done in universal_robots_hand_eye_script.urp on the robot.
+
+Further explanation of this sample is found in our knowledge base:
+https://zivid.atlassian.net/wiki/spaces/ZividKB/pages/103481384/
+"""
+import argparse
+from pathlib import Path
+import time
+import datetime
+
+import cv2
+import numpy as np
+from scipy.spatial.transform import Rotation as R
+import zivid
+import rtde.rtde as rtde
+import rtde.rtde_config as rtde_config
+
+
+def _options():
+    """Function for taking in arguments from user
+
+    Returns:
+        Arguments from user
+
+    """
+    parser = argparse.ArgumentParser(
+        description="full_hand_eye.py [-h] [--eih] [--eth] [--ip <IP address>]",
+        formatter_class=argparse.RawTextHelpFormatter,
+    )
+    parser.add_argument(
+        "--mode",
+        help="Type of calibration",
+        choices=["eih", "eye-in-hand", "eth", "eye-to-hand"],
+        required=True,
+    )
+    parser.add_argument("--ip", help="IP address to robot", required=True)
+
+    return parser.parse_args()
+
+
+def _write_robot_state(
+    con: rtde, input_data, finish_capture: bool = False, camera_ready: bool = False
+):
+    """Write to robot I/O registrer
+
+    Args:
+        con: Connection between computer and robot
+        input_data: Input package containing the specific input data registers
+        finish_capture: Boolean value to robot_state that q_r scene capture is finished
+        camera_ready: Boolean value to robot_state that camera is ready to capture images
+    """
+
+    if finish_capture:
+        input_data.input_bit_register_64 = 1
+    else:
+        input_data.input_bit_register_64 = 0
+
+    if camera_ready:
+        input_data.input_bit_register_65 = 1
+    else:
+        input_data.input_bit_register_65 = 0
```

Comment:
This could be simplified a lot, since `int(True)` returns 1 and `int(False)` returns 0.
The whole thing can be:
```
input_data.input_bit_register_64 = int(finish_capture)
input_data.input_bit_register_65 = int(camera_ready)
```